<launch>

    <!-- ############################ Tool approach offsets ############################ -->
    <!-- distance in meters from the face to the tool when begining approach -->
    <arg name="dist_face_start" value="0.08"/>
    <!-- distance in meters from the face to the tool to stop -->
    <arg name="dist_face_end" value="0.01"/>
    <!-- distance in meters from the wrist to the tool tip -->
    <arg name="tool_dist" value="0.110"/>
    <!-- ################################################################################# -->

    <!-- ############################ Force torque parameters ############################ -->
    <node pkg="pr2_collision_monitor" type="force_torque_monitor" name="force_torque_monitor" output="screen" respawn="true">
        <param name="force_threshold" type="double" value="0.0005" />
        <param name="time_constant" type="double" value="0.001" />
        <param name="history" type="int" value="500" />
        <remap from="/force_topic" to="/force_torque_ft3_Vec3"/>
    </node>
    <!-- ################################################################################# -->

    <include file="$(find pr2_arm_kinematics)/launch/pr2_ik_rarm_node.launch"/>
    <include file="$(find pr2_arm_kinematics)/launch/pr2_ik_larm_node.launch"/>
    <include file="$(find hrl_arm_move_behaviors)/launch/epc_move_actionlib.launch"/>

    <node pkg="tf" type="static_transform_publisher" name="l_scratcher_tip_frame" args="$(arg tool_dist) 0 0 0 0 0 /l_gripper_tool_frame /l_scratcher_tip 100" />
    <node pkg="tf" type="static_transform_publisher" name="l_scratcher_touch_frame" args="-$(arg dist_face_end) 0 0 0 0 0 /l_scratcher_tip /l_scratcher_touch 100" />
    <node pkg="tf" type="static_transform_publisher" name="l_scratcher_approach_frame" args="$(arg dist_face_start) 0 0 0 0 0 /l_scratcher_tip /l_scratcher_approach 100" />

    <node pkg="tf" type="static_transform_publisher" name="r_scratcher_tip_frame" args="$(arg tool_dist) 0 0 0 0 0 /r_gripper_tool_frame /r_scratcher_tip 100" />
    <node pkg="tf" type="static_transform_publisher" name="r_scratcher_touch_frame" args="-$(arg dist_face_end) 0 0 0 0 0 /r_scratcher_tip /r_scratcher_touch 100" />
    <node pkg="tf" type="static_transform_publisher" name="r_scratcher_approach_frame" args="$(arg dist_face_start) 0 0 0 0 0 /r_scratcher_tip /r_scratcher_approach 100" />

</launch>
